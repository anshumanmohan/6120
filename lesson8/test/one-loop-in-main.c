int add(int a, int b)
{
  return a + b;
}

int main()
{
  int a = 0;
  for (; a < 100;)
  {
    int b = add(3, 3);
    int c = 3 + 3;
    a = a + b + c;
  }
  return a;
}