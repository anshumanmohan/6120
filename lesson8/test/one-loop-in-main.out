I saw a function called add
	the function has 0 loop(s) total
I saw a function called main
	I'm a block inside loop #1
		I'm an instruction inside that block
		I'm an instruction inside that block
		I'm an instruction inside that block
	I'm a block inside loop #1
		I'm an instruction inside that block
		I'm an instruction inside that block
		I'm an instruction inside that block
		I'm an instruction inside that block
		I'm an instruction inside that block
		I'm an instruction inside that block
		I'm an instruction inside that block
	the function has 1 loop(s) total
